GVFS/GVFS.Hooks/Program.cs

```diff
@@ -395,6 +395,7 @@ private static bool ShouldLock(string[] args)
                 case "difftool":
                 case "fetch":
                 case "for-each-ref":
+                case "git-submodule":

```

Comment:
> submodule should be blocked in git

I believe the block in git is happening after git acquires the GVFSLock, and so if `submodule` is run during a build it can break the build because VFS4G will think a git command is running and not allow placeholders to be created.

Not sure about `submodule` vs `git-submodule` though

Reply:
This is still in draft :) I mainly posted this to get the test runs while I stepped away. I was going to add more details and options before taking it out of draft mode.

But @wilbaker is correct with his assessment.

Reply:
If that is the case then don't we need all the commands that are blocked in git in this list to not take the lock?

Reply:
Yes, if this is the approach we want to take, that would be a more complete change.

Reply:
@kewillford - do you know of a good reference list of blocked git commands?

Reply:
[This commit](https://github.com/Microsoft/git/commit/45523ec7e83c52d5a3812a50c31c5f6ed1c9a0cd) is what is being blocked in git.  I'm not sure we should add them to the list that doesn't lock because if we for some reason decide that we can unblock we might need the command to take the lock or at least test the command and we should probably default to taking the lock.  Also it would make more sense if we know that git will block the command to just block the command in the hook like we were before. What are your thoughts?